tests/script/test_script.py

```diff
@@ -0,0 +1,133 @@
+from functools import partial
+import json
+import pytest
+import sys
+from unittest import mock
+
+from yapapi import WorkContext
+from yapapi.events import CommandExecuted
+from yapapi.script import Script
+from yapapi.script.command import Deploy, Start
+
+
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="AsyncMock requires python 3.8+")
+class TestScript:
+    @pytest.fixture(autouse=True)
+    def setUp(self):
+        self._on_download_executed = False
+
+    @pytest.fixture
+    def work_context(self):
+        return WorkContext(mock.MagicMock(), mock.MagicMock(), storage=mock.AsyncMock())
```

Comment:
hmm, I think this could be converted into a factoryboy factory so that it's reusable anywhere else where we'd like to have a mock WorkContext....